libsql/internal/http/http.go

```diff
@@ -3,61 +3,90 @@ package http
 import (
 	"bytes"
 	"context"
+	"database/sql/driver"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
+	"sort"
 	"time"
 
 	"github.com/xwb1989/sqlparser"
 )
 
 var httpClient = &http.Client{Timeout: 120 * time.Second}
 
-type params struct {
-	Names  []string
-	Values []any
-}
+type paramsType int
 
-func (p *params) MarshalJSON() ([]byte, error) {
-	if len(p.Values) == 0 {
-		return json.Marshal([]string{})
-	}
-	if len(p.Names) == 0 {
-		return json.Marshal(p.Values)
+const (
+	namedParameters paramsType = iota
+	positionalParameters
+)
+
+type parameters struct {
+	paramsValues []struct {
+		name  string
+		value any
 	}
-	m := map[string]interface{}{}
-	for idx := range p.Values {
-		m["@"+p.Names[idx]] = p.Values[idx]
+	paramsType paramsType
```

Comment:
This is not needed. One can check paramsValues[0].name != "" to check if those params are named or not.

Reply:
I did this way because I think this way it's clear that all parameters has the same types. What do you think?

Reply:
I think in this case you can add a comment on the type explaining this. If there's a way to tell something using the existing state, I wouldn't add another redundant field. A method like `named() bool` should be enough.